### PR TITLE
Extracted duplicate code

### PR DESCRIPTION
**Pull request by StefanSpieker**
I extracted duplicated code in an unit test and used this function within the class and merged two switch cases in the class under test.

### Proposed changelog entries

No changelog entry needed

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 1]
```diff
@@ -22,7 +22,15 @@
...
-        String password = null;
+        assertkeyPair(file, null);
+    }
+
+    /**
+     * Extracted method to assert the keyPair private and public are not null.
+     * @param file the file to load the key from
+     * @param password the password
+     */
+    private void assertkeyPair(File file, String password) throws IOException, GeneralSecurityException {
```
Comment (jvz) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]: Naming consistency nit:

```suggestion
    private void assertKeyPair(File file, String password) throws IOException, GeneralSecurityException {
```

**Review by oleg-nenashev — APPROVED** [review]

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 2]
```diff
@@ -22,7 +22,15 @@
     @Test
     public void loadKeyDSA() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("dsa").getFile());
-        String password = null;
+        assertkeyPair(file, null);
```
Comment (oleg-nenashev) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
        assertkeyPair(file, null);
```
```suggestion
        assertkeyPair(file, null);
```
```suggestion
        assertKeyPair(file, null);
```

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 3]
```diff
@@ -48,11 +53,7 @@ public void loadKeyDSAPassword() throws IOException, GeneralSecurityException {
     @Test
     public void loadKeyRSA() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("rsa").getFile());
-        String password = null;
-        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
-        assertNotNull(keyPair);
-        assertNotNull(keyPair.getPrivate());
-        assertNotNull(keyPair.getPublic());
+        assertkeyPair(file, null);
```
Comment (oleg-nenashev) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
        assertKeyPair(file, null);
```

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 4]
```diff
@@ -36,10 +44,7 @@ public void loadKeyDSA() throws IOException, GeneralSecurityException {
     public void loadKeyDSAPassword() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("dsa-password").getFile());
         String password = "password";
-        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
-        assertNotNull(keyPair);
-        assertNotNull(keyPair.getPrivate());
-        assertNotNull(keyPair.getPublic());
+        assertkeyPair(file, password);
```
Comment (oleg-nenashev) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
        assertKeyPair(file, password);
```

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 5]
```diff
@@ -62,10 +63,7 @@ public void loadKeyRSA() throws IOException, GeneralSecurityException {
     public void loadKeyRSAPassword() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("rsa-password").getFile());
         String password = "password";
-        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
-        assertNotNull(keyPair);
-        assertNotNull(keyPair.getPrivate());
-        assertNotNull(keyPair.getPublic());
+        assertkeyPair(file, password);
```
Comment (oleg-nenashev) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
        assertKeyPair(file, password);
```

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 6]
```diff
@@ -74,11 +72,7 @@ public void loadKeyRSAPassword() throws IOException, GeneralSecurityException {
     @Test
     public void loadKeyOpenSSH() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("openssh").getFile());
-        String password = null;
-        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
-        assertNotNull(keyPair);
-        assertNotNull(keyPair.getPrivate());
-        assertNotNull(keyPair.getPublic());
+        assertkeyPair(file, null);
```
Comment (oleg-nenashev) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
        assertKeyPair(file, null);
```

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
```
13:36:10  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project cli: Compilation failure: Compilation failure: 
13:36:10  [ERROR] /home/jenkins/workspace/Core_jenkins_PR-4206/cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java:[25,8] error: cannot find symbol
13:36:10  [ERROR]   symbol:   method assertkeyPair(File,<null>)
13:36:10  [ERROR]   location: class PrivateKeyProviderTest
```

**Comment by StefanSpieker** [comment]
Sorry, I just applied the requested changes without noticing that all methods had to be changed too. Thanks for the suggested changes.

**Review by oleg-nenashev — APPROVED** [review]
No worries, and thanks for fixing it!

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 7]
```diff
@@ -22,7 +22,15 @@
     @Test
     public void loadKeyDSA() throws IOException, GeneralSecurityException {
         File file = new File(this.getClass().getResource("dsa").getFile());
-        String password = null;
+        assertKeyPair(file, null);
+    }
+
+    /**
+     * Extracted method to assert the keyPair private and public are not null.
```
Comment (timja) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
     * Asserts the keyPair private and public are not null.
```

File: cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java [hunk 8]
```diff
@@ -22,7 +22,15 @@
...
+     * @param file the file to load the key from
+     * @param password the password
+     */
+    private void assertKeyPair(File file, String password) throws IOException, GeneralSecurityException {
```
Comment (timja) [inline cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]:
```suggestion
    private void assertKeyPairNotNull(File file, String password) throws IOException, GeneralSecurityException {
```

Comment (StefanSpieker) [reply cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java]: commited the change as suggested

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
I plan to merge it tomorrow if no negative feedback

**Pull request merged**